### DISCUSSION TITLE
How to modify the MAX device to accept more than 32 parameters?

### DISCUSSION THREAD

**Discussion by rds2024-3004** [zsteinkamp/m4l-zs-Knobbler3#7] [General]
Hello. I'm new to M4L but already know how to do the TouchOSC side. Could someone help me extend the available parameters to more than 32 by editing the MAX device?

Thanks

**Comment by zsteinkamp** [comment]
Heya @rds-2024-3004 I'll look into it. What number are you looking to increase that to? I'm assuming you're making additional tabs in the TouchOSC interface?

**Reply by rds2024-3004** [reply]
I'm looking for about 64 per M4L device. I am currently customizing the touchOSC template. So I have 32 parameters per tab. I could extend the numbers my own by using other connections but it would overlap with my other controls. Also, is there a way to assign the refresh button to a midi cc? 

Thanks a lot! I really love this project.

**Reply by rds2024-3004** [reply]
Sorry I have another question but a different topic as I am playing around with this. Is there a way include knobbler in a group and save it as template? So that I don't have to redo the mappings every time I load my plugin. 

Upon trying by saving it in a group template/preset, the mappings are gone after loading the preset.
